core/controllers/topics_and_skills_dashboard.py

```diff
@@ -166,6 +166,107 @@ def get(self):
         self.render_json(self.values)
 
 
+class CategorizedAndUntriagedSkillsDataHandler(base.BaseHandler):
```

Comment:
If there's existing code that does a similar computation, try to deduplicate. Put the core logic in a common place that can be used by all the cases which need this functionality. 

Ideally, do not, ever, have two functions doing basically the same thing. I know it's easier to just write an additional separate thing -- but that ends up being more code to maintain, and increases the chance of bugs and discrepancies, so it is not a good practice, especially in a large codebase.

Reply:
I have shifted the logic to `skill_services.py` now and using the same logic for both the places where it is required. PTAL.

Reply:
Thanks. I have a question: could you explain what the difference is between what the topics_and_skills_dashboard page needs and what the exp editor page needs? I want to ensure that for the latter we are returning only the minimal thing that is needed. (We should only deduplicate that part and not everything.)

Reply:
The topics_and_skills_dashboard needs all of the following:
```
untriaged_skill_summary_dicts, mergeable_skill_summary_dicts, topic_summary_dicts, total_skill_count, all_classroom_names, can_delete_topic, can_create_topic, can_delete_skill, can_create_skill, categorized_skills_dict
```
And the exploration editor just needs the ids and descriptions of all the skills so that it can the skill selector component can work properly.

Reply:
Currently, for categorized skills, just the ids and descriptions are sent to the frontend but for untriaged skills, the whole skill_summary _dict is being sent (id, description and other properties). We can untriaged skills data more but I think it would require a lot of changes in the frontend. Should I do it in this PR itself or raise another PR to minify the data (after this is merged)?

Reply:
I'd say this PR. Let's fix and standardize everything once and for all so that it doesn't cause confusion for any developer who works on this.

You might also want to introduce a domain object for the dict with "skill id" and "description", since that is becoming a standard concept.